### PR TITLE
keptn/keptn#4404 Auto update shkeptnspecversion on release of new keptn/spec version

### PR DESCRIPTION
**Pull request by mowies**
**This PR**
* adds a new pipeline that gets triggered by a release in the keptn/spec repo
* remove the `version` file since it's not used anymore
* adds a yaml config to adjust the `shkeptnspecversion` that is sent per default with every cloud event
* adjust unit tests accordingly

Fixes keptn/keptn#4404

Signed-off-by: Moritz Wiesinger <moritz.wiesinger@dynatrace.com>

### CODE REVIEW

File: .github/workflows/auto-update-keptn-spec.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        uses: actions/checkout@v2
+
+      - name: Extract spec tag
+        run: echo "SPEC_TAG=${SPEC_REF#refs/tags/}" >> $GITHUB_ENV
+
+      - name: Update keptn/spec
+        env:
+          SPEC_TAG: ${{ env.SPEC_TAG }}
+        run: |
+          echo 'shkeptnspecversion: "$SPEC_TAG"\n' > ./config/config.yaml
```
Comment (warber) [inline .github/workflows/auto-update-keptn-spec.yml:26]: not completely sure, but I think `\n` is printed as string into the file... pls check

Comment (mowies) [reply .github/workflows/auto-update-keptn-spec.yml:26]: it works on my machine :)
but actually echo already adds a newline anyways so the additional `\n` can be removed

File: config/config.yaml [hunk 2]
```diff
@@ -0,0 +1 @@
+shkeptnspecversion: "0.2.1"
```
Comment (warber) [inline config/config.yaml:1]: i think the current spec version is 0.2.3, let's change it to this, to have the correct one on master after merging

Comment (mowies) [reply config/config.yaml:1]: sure I will update it

Comment (mowies) [reply config/config.yaml:1]: but we could also take this opportunity to properly test the new pipeline after merging this PR

**Review by warber — APPROVED** [review]

**Pull request merged**